eduiddashboard/views/nins.py

```diff
@@ -523,13 +523,12 @@ def get_template_context(self):
 
         settings = self.request.registry.settings
         enable_mm = settings.get('enable_mm_verification')
-        session_user = get_session_user(self.request, raise_on_not_logged_in = False)
 
         context.update({
-            'nins': session_user.nins.to_list_of_dicts(),
-            'not_verified_nins': get_not_verified_nins_list(self.request, session_user),
+            'nins': self.user.nins.to_list_of_dicts(),
+            'not_verified_nins': get_not_verified_nins_list(self.request, self.user),
             'active_nin': self.get_active_nin(),
-            'has_mobile': has_confirmed_mobile(session_user),
+            'has_mobile': has_confirmed_mobile(self.user),
```

Comment:
I don't know the code flow of the application here. None of the options sounds too good to me actually :).

Using the user in the context is like using global variables, and we've had a number of problems come from that.

Fetching the user using get_session_user sounds like it might become a race condition if I understand what happens here. Sounds like we want to update the user and render a page with the users new data, but what happens if we reach the "render a page" before the updates have propagated to the central userdb?

If I'm correct about what's being done here, the code needs restructuring IMO. But today might not be the day to do that. To answer your question, if it works to call get_session_user